spec/system/add_search_rspec.rb

```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe 'Search function', type: :system do
+  before do
+    @ruby_programming_language = create :programming_language, name: 'Ruby'
+    create :programming_language, name: 'Javascript'
+    create :programming_language, name: 'Php'
+
+    @ruby_devloper = create :developer, email: 'phan.nguyen@example.com'
+    @ruby_devloper.developer_programming_languages.create(
+      programming_language: @ruby_programming_language
+    )
```

Comment:
The lines (12 ~ 14) look redundant.
You can pass `ruby_programming_language` in line 11, can't you?
I didn't check the following code, but something like this.
```
@ruby_devloper = create :developer, email: 'phan.nguyen@example.com', programming_languages: [@ruby_programming_language]
```

Reply:
Inserting a record in an intermediate table directly, `developer_programming_languages` this case, looks wrong to me (I might be wrong).